src/neuroconv/tools/nwb_helpers/_configuration_models/_base_dataset_io.py

```diff
@@ -298,7 +304,7 @@ def from_neurodata_object(
                 number_of_channels=number_of_channels, number_of_frames=number_of_frames, dtype=dtype
             )
 
-            buffer_shape = None  # This is the non-iterative path
+            buffer_shape = full_shape
```

Comment:
why did we change this?

Reply:
See https://github.com/catalystneuro/neuroconv/pull/1003#discussion_r2487885896

Reply:
This comment lead me nowhere. Can you double check?

Reply:
How about this [one](https://github.com/catalystneuro/neuroconv/pull/1003/files#r2516903381)?